app.rb

```diff
@@ -43,6 +43,19 @@
 end
 
 #
+#  Fetches a random definition by following the /random.php redirect
+#  and taking out the term with a regexp
+#
+get '/random' do
+  rdef = ud.get_random_definition()
+    unless valid_response? definition
+    raise NoDefinition, params[:term]
+  end
+
+  jsonp definition
```

Comment:
You are most likely correct, since I have not solid knowledge of Ruby I'm in no position to discuss this. This way, feel free to change this at will.

Is it doing some error handling with valid_response?